compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java

```diff
@@ -624,7 +624,7 @@ public boolean isNotUnknownTypeRef(BLangUserDefinedType td) {
         Location pos = td.pos;
         LocationData locationData = new LocationData(td.typeName.value, pos.lineRange().startLine().line(),
                 pos.lineRange().startLine().offset());
-        return unknownTypeRefs.add(locationData);
+        return dlog.isMute() || unknownTypeRefs.add(locationData);
```

Comment:
Will this reflect the function signature?

Reply:
+1, I think we have to add this change where `isNotUnknownTypeRef` is called. 

I suggest checking whether we need a map like this `unknownTypeRefs`. This map is used since we called the same parent method at different times. We need to check the possibility of optimizing those calls.

Reply:
@KavinduZoysa @nipunayf Is the suggested change done or are we good to merge this without it?